mmv1/provider/core.rb

```diff
@@ -190,12 +190,72 @@ def compile_file_list(output_folder, files, file_template, pwd = Dir.pwd)
           Google::LOGGER.debug "Compiling #{source} => #{target}"
           file_template.generate(pwd, source, target, self)
 
+          add_hashicorp_copyright_header(output_folder, target)
           replace_import_path(output_folder, target)
         end
       end.map(&:join)
       Dir.chdir pwd
     end
 
+    def add_hashicorp_copyright_header(output_folder, target)
+      # only add copyright headers when generating TPG and TPGB
+      return unless output_folder.end_with?('terraform-provider-google') ||
+                    output_folder.end_with?('terraform-provider-google-beta')
+
+      # Prevent adding copyright header to files with paths or names matching the strings below
+      # NOTE: these entries need to match the content of the .copywrite.hcl file originally
+      #       created in https://github.com/GoogleCloudPlatform/magic-modules/pull/7336
+      ignored_folders = [
+        '.github/',
+        '.release/',
+        '.changelog/',
+        'examples/',
+        'META.d/'
+      ]
+      ignored_files = [
+        'go.mod',
+        '.goreleaser.yml',
+        '.golangci.yml',
+        'terraform-registry-manifest.json'
```

Comment:
Should `release-metadata.hcl.erb` be added to `ignored_files`? This file caused the build failure.

Reply:
Which build failure was/is that?

The `release-metadata.hcl` file in the downstream is in the `.release/` folder so it is ignored based on its location, and that's how the file is ignored in [the .copywrite.hcl file](https://github.com/hashicorp/magic-modules/blob/534bbabd4f2008e62180f5cbf27fadc2805a947e/mmv1/third_party/terraform/.copywrite.hcl.erb). This means that the automation that wants to add headers will not assert the presence of the headers in files within  the `.release/` directory and this code won't add the headers in those files also.

Reply:
I notice the check   `presubmit-generate-diffs` failed because of this file.

https://pantheon.corp.google.com/cloud-build/builds/f04fe851-7928-4070-aa58-f7e04367e276?project=graphite-docker-images

Reply:
I'm afraid access to that resource is limited to Googlers - could you please summarise the problem? Is it currently ongoing?

Reply:
Ah it's also the CloudBuild checks on this PR- I'll go check

Reply:
Yes, it is the check presubmit-generate-diffs on this PR.

Reply:
I'll update the `language_from_filename` function that is the source of that problem